tests/hostnameshardtests/ingress_annotation_test.go

```diff
@@ -0,0 +1,421 @@
+		HostNames:   []string{"v1"},
+		ServiceName: "avisvc",
+	}).Ingress()
+	ann := make(map[string]string)
+	ann[lib.PassthroughAnnotation] = "true"
+	ingrFake.SetAnnotations(ann)
+	_, err := KubeClient.NetworkingV1beta1().Ingresses("default").Create(context.TODO(), ingrFake, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("error in adding Ingress: %v", err)
+	}
+	integrationtest.PollForCompletion(t, DefaultPassthroughModel, 5)
+
+	aviModel := ValidatePassthroughModel(t, g, DefaultPassthroughModel)
+	graph := aviModel.(*avinodes.AviObjectGraph)
+	vs := graph.GetAviVS()[0]
+	VerifyPasthrough(t, g, vs)
+
+	g.Expect(vs.PassthroughChildNodes).To(gomega.HaveLen(1))
+	passInsecureNode := vs.PassthroughChildNodes[0]
+	g.Expect(passInsecureNode.Name).To(gomega.Equal("cluster--Shared-Passthrough-0-insecure"))
+	g.Expect(passInsecureNode.HttpPolicyRefs[0].RedirectPorts[0].Hosts[0]).To(gomega.Equal("foo.com"))
+	g.Expect(passInsecureNode.HttpPolicyRefs[0].RedirectPorts[0].StatusCode).To(gomega.Equal("HTTP_REDIRECT_STATUS_CODE_302"))
+
+	err = KubeClient.NetworkingV1beta1().Ingresses("default").Delete(context.TODO(), passthroughIngressName, metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatalf("error in deleting Ingress: %v", err)
+	}
+	VerifyPassthroughIngressDeletion(t, g, DefaultPassthroughModel, 0, 0)
+	TearDownTestForIngress(t, DefaultPassthroughModel)
+}
+
+func TestPassthroughIngressRemoveAnnotation(t *testing.T) {
+	g := gomega.NewGomegaWithT(t)
+	SetUpTestForIngress(t, DefaultPassthroughModel)
+	ingrFake := (integrationtest.FakeIngress{
+		Name:        passthroughIngressName,
+		Namespace:   "default",
+		DnsNames:    []string{"foo.com"},
+		Ips:         []string{"8.8.8.8"},
+		HostNames:   []string{"v1"},
+		ServiceName: "avisvc",
+	}).Ingress()
+	ann := make(map[string]string)
+	ann[lib.PassthroughAnnotation] = "true"
+	ingrFake.SetAnnotations(ann)
+	_, err := KubeClient.NetworkingV1beta1().Ingresses("default").Create(context.TODO(), ingrFake, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("error in adding Ingress: %v", err)
+	}
+	integrationtest.PollForCompletion(t, DefaultPassthroughModel, 5)
+
+	aviModel := ValidatePassthroughModel(t, g, DefaultPassthroughModel)
+	graph := aviModel.(*avinodes.AviObjectGraph)
+	vs := graph.GetAviVS()[0]
+	VerifyPasthrough(t, g, vs)
+
+	ann = make(map[string]string)
+	ingrFake.SetAnnotations(ann)
+	ingrFake.ResourceVersion = "2"
+	_, err = KubeClient.NetworkingV1beta1().Ingresses("default").Update(context.TODO(), ingrFake, metav1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("error in updating Ingress: %v", err)
+	}
+	VerifyPassthroughIngressDeletion(t, g, DefaultPassthroughModel, 0, 0)
+
+	err = KubeClient.NetworkingV1beta1().Ingresses("default").Delete(context.TODO(), passthroughIngressName, metav1.DeleteOptions{})
+	if err != nil {
+		t.Fatalf("error in adding Ingress: %v", err)
+	}
+	TearDownTestForIngress(t, DefaultPassthroughModel)
+}
+
+func TestPassthroughIngressAddAnnotation(t *testing.T) {
```

Comment:
@monotosh-avi : Should we add a test case where two hosts maps to single shared VS to check multiple DSset creation and only single redirect httpps.? Does split traffic allowed in k8 implementation of passthrough as allowed in openshift?

Reply:
Good idea Akshay. Added a test to check multiple ingress case. 
Unlike openshift routes, Ingresses don't have alternate backends, we are not adding that functionality here.